addon/-legacy-private/system/relationships/state/has-many.js

```diff
@@ -16,25 +17,36 @@ export default class ManyRelationship extends Relationship {
     // we create a new many array, but in the interim it will be updated if
     // inverse internal models are unloaded.
     this._retainedManyArray = null;
-    this._loadingPromise = null;
+    this._promiseProxy = null;
     this._willUpdateManyArray = false;
     this._pendingManyArrayUpdates = null;
   }
 
-  _updateLoadingPromise(promise, content) {
-    if (this._loadingPromise) {
-      if (content) {
-        this._loadingPromise.set('content', content);
-      }
-      this._loadingPromise.set('promise', promise);
-    } else {
-      this._loadingPromise = PromiseManyArray.create({
-        promise,
-        content,
-      });
+  get currentState() {
+    return this.members.list;
+  }
+
+  get hasRelatedResources() {
```

Comment:
Can you please write a comment describing the intended behavior of this method

Reply:
This seems to be maybe equivalent to the `hasLoaded` property? If i'm reading this correctly, it should return true if you have an empty array, in which case the naming is pretty misleading

Reply:
changed to `allInverseRecordsAreLoaded`